onnxruntime/core/providers/cuda/tensor/tile.cc

```diff
@@ -58,27 +59,47 @@ Status Tile<T>::ComputeInternal(OpKernelContext* ctx) const {
     fdm_output_strides[i] = fast_divmod(static_cast<int>(output_pitches[i]));
   }
 
+  static_assert(sizeof(float) == sizeof(int32_t), "Float and Int32 are of different sizes");
+  static_assert(sizeof(double) == sizeof(int64_t), "Double and Int64 are of different sizes");
+
   if (output_tensor.Shape().Size() > 0) {
-    TileImpl(
-        rank,
-        fdm_input_shape,
-        input_strides,
-        reinterpret_cast<const typename ToCudaType<T>::MappedType*>(input_data),
-        fdm_output_strides,
-        reinterpret_cast<typename ToCudaType<T>::MappedType*>(output_data),
-        output_tensor.Shape().Size());
+    if (input_tensor.IsDataType<float>() ||
+        input_tensor.IsDataType<int32_t>()) {
```

Comment:
Tests already exist for these types (they were testing the CPU kernel, now they will test the CUDA kernel too)